Bottom: 6e1dcf855b0e9d153305fa82245ee1848bc53ca9
Top:    82bdfe8723d136377930b6631027ad3c27f4f70b
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-01-22 18:16:59 +0800

Refresh of vhost-descriptor-batch

---

diff --git a/drivers/vhost/vhost.c b/drivers/vhost/vhost.c
index 33ac2b1..1a496d9 100644
--- a/drivers/vhost/vhost.c
+++ b/drivers/vhost/vhost.c
@@ -440,6 +440,7 @@ void vhost_dev_init(struct vhost_dev *dev,
 		vq->indirect = NULL;
 		vq->heads = NULL;
 		vq->dev = dev;
+		vq->descq.head = vq->descq.tail = 0;
 		mutex_init(&vq->mutex);
 		vhost_vq_reset(dev, vq);
 		if (vq->handle_kick)
diff --git a/drivers/vhost/vhost.h b/drivers/vhost/vhost.h
index 807828a..9a188fe 100644
--- a/drivers/vhost/vhost.h
+++ b/drivers/vhost/vhost.h
@@ -86,6 +86,7 @@ enum vhost_uaddr_type {
 
 struct vhost_vring_desc {
 	struct vring_desc descs[64];
+	__virtio16 indices[64];
 	int head;
 	int tail;
 };
